### PR TITLE
feat: add admin api reports visitors

### LINKED ISSUE #26: [v2] Add /admin-api/reports/visits endpoint for paginated visitor attendance reports
Implement a new reporting endpoint that returns paginated visitor attendance information aggregated per visitor per day for a given date range. The endpoint should support optional filters and return sign-in / sign-out timestamps, counts and duration based on the provided SQL logic.

⸻

Endpoint

`GET /admin-api/reports/visits`

⸻

Request parameters

**Required**
	•	`from` (date, YYYY-MM-DD) — start date (inclusive). Minimal precision: day.
	•	`to` (date, YYYY-MM-DD) — end date (inclusive).

_Optional_
	•	`is_student` (boolean) — filter by student flag.
	•	`year_group` (integer) — filter by year group.
	•	`sign_status` (string) — filter by sign status (not_signed, signed_in, signed_out).
	•	`page` (integer, default 1) — page number.
	•	`limit` (integer, default 20) — records per page.

Date range parameters are day-precise. Convert to to end-of-day in queries (e.g. to + 1 day exclusive) or use inclusive date logic.

Response format (JSON)

```
{
  "page": 1,
  "limit": 20,
  "total": 154,
  "data": [
    {
      "visit_date": "2026-03-10",
      "visitor_id": 25,
      "name": "John",
      "surname": "Smith",
      "is_student": true,
      "year_group": 10,
      "visits_count": 2,
      "sign_status": "signed_out",
      "signed_in": "2026-03-10T08:45:12Z",
      "signed_out": "2026-03-10T15:12:55Z",
      "duration_minutes": 387
    },
    {
      "visit_date": "2026-03-10",
      "visitor_id": 31,
      "name": "Anna",
      "surname": "Brown",
      "is_student": true,
      "year_group": 9,
      "visits_count": 1,
      "sign_status": "signed_in",
      "signed_in": "2026-03-10T09:02:33Z",
      "signed_out": null,
      "duration_minutes": null
    }
  ]
}

```

⸻

Sign status logic
	•	`not_signed` — visits_count = 0
	•	`signed_in` — visits_count is odd (1,3,…) — visitor currently signed in
	•	`signed_out` — visits_count is even (2,4,…) — visitor signed out

⸻

Acceptance criteria
	•	Endpoint GET /admin-api/reports/visits implemented and documented.
	•	from and to date parameters are required and interpreted at day granularity.
	•	Results are paginated (page, limit) and include total count.
	•	Filters is_student, year_group, sign_status apply correctly.
	•	Response fields match the defined response format (including signed_in, signed_out, duration_minutes).
	•	Sign-in/out timestamps and duration_minutes calculated per the SQL logic.
	•	Endpoint performs efficiently for large datasets (no full-table cross-joins without justification).
	•	Unit/integration tests cover core logic and edge cases (no visits, single visit, multiple visits, filters).
	•	API documentation (OpenAPI/Swagger) updated.

⸻

Implementation notes / suggestions
	•	Convert from/to into [start_datetime, end_datetime) for SQL (e.g. from 00:00:00 to to + 1 day 00:00:00 exclusive).
	•	Avoid CROSS JOIN visitors for production if the visitors table is large — instead: generate the date series and join only visitors who have tracks in the period, or pre-aggregate tracker.track by date and visitor then join to visitors.
	•	Add indexes on tracker.track(visitor_id, created_at) and on visitors(id) if not present.
	•	Consider adding csv/excel export option for reports (future extension).
	•	Return timestamps in ISO-8601 UTC (or include timezone explicitly).

⸻

If you want, I can also generate an OpenAPI snippet (YAML) and a basic controller skeleton you can paste into the repo.

### PR DESCRIPTION
**Pull request by buzyka**
closes #26

### CODE REVIEW

File: internal/usecase/adminapi/reports_visits.go [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+		o(opts)
+	}
+
+	// Validate search params
+	if err := a.validateReportsVisitsRequest(&fromTime, &toTime, opts); err != nil {
+		return nil, err
+	}
+
+	if toTime.Before(fromTime) {
+		toTime, fromTime = fromTime, toTime
```
[the remark this answers is not in the document]
Comment (buzyka) [reply internal/usecase/adminapi/reports_visits.go:106]: it is intended behaviour
